### ISSUE TITLE
perf: unlock without defer

### ISSUE THREAD

**Issue by trim21** [masaushi/accessory#18]
defer is kind of expensive in this simple variable assignment case

**Comment by masaushi** [comment]
Thank you @trim21 for your comment!

I think you're right, so I'm considering to fix locking without defer.

**Comment by trim21** [comment]
any news about this issue?

**Comment by masaushi** [comment]
I'm afraid there hasn't been any progress so far.

**Comment by lrstanley** [comment]
I don't think this is actually needed. It used to be the case that there were a lot of edge cases where defer wasn't inlined correctly, and thus could have (slightly) worse performance when unlocking a mutex with defer, however, the Go compiler in 95% of situations should inline it correctly now so it's not technically a defer in the typical sense.
